sway-error/src/warning.rs

```diff
@@ -84,7 +84,14 @@ pub enum Warning {
     DeadStorageDeclarationForFunction {
         unneeded_attrib: String,
     },
-    MatchExpressionUnreachableArm,
+    MatchExpressionUnreachableArm {
+        match_value: Span,
+        preceding_arms: Span,
+        preceding_arm_is_catch_all: bool,
```

Comment:
Perhaps a name like `follows_catch_all_arm` would be more clear

Reply:
It's definitely more clear! I'll rename it in the upcoming PR.

Reply:
Hmm, thinking of `Either` right now :thinking:  That would actually properly solve both comments of yours: either preceding arm*s* or a single catch all arm, and no need for additional boolean flag... I'll think of it.